### ISSUE TITLE
fuzz: UndefinedBehaviorSanitizer warnings in netbase.cpp

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#21479]
https://cirrus-ci.com/task/6113700391157760:

```
Run netbase_dns_lookup with args ['/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz', '-runs=1', '/tmp/cirrus-ci-build/ci/scratch/qa-assets/fuzz_seed_corpus/netbase_dns_lookup']INFO: Seed: 2191391191
INFO: Loaded 1 modules   (515777 inline 8-bit counters): 515777 [0x55f8604c4498, 0x55f860542359), 
INFO: Loaded 1 PC tables (515777 PCs): 515777 [0x55f860542360,0x55f860d20f70), 
INFO:      510 files found in /tmp/cirrus-ci-build/ci/scratch/qa-assets/fuzz_seed_corpus/netbase_dns_lookup
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: seed corpus: files: 510 min: 1b max: 4042b total: 273870b rss: 117Mb
netbase.cpp:212:37: runtime error: implicit conversion from type 'int' of value -2147483632 (32-bit, signed) to type 'uint16_t' (aka 'unsigned short') changed the value to 16 (16-bit, unsigned)
    #0 0x55f85dbd91cc  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x338e1cc)
    #1 0x55f85cfd0f5f  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x2785f5f)
    #2 0x55f85ccebeb7  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x24a0eb7)
    #3 0x55f85e498067  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x3c4d067)
    #4 0x55f85e497d15  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x3c4cd15)
    #5 0x55f85cbf1851  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a6851)
    #6 0x55f85cbf0f95  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a5f95)
    #7 0x55f85cbf38b7  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a88b7)
    #8 0x55f85cbf3c19  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a8c19)
    #9 0x55f85cbe28ee  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23978ee)
    #10 0x55f85cc0b732  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23c0732)
    #11 0x7f05b2c830b2  (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #12 0x55f85cbb768d  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x236c68d)
SUMMARY: UndefinedBehaviorSanitizer: implicit-signed-integer-truncation netbase.cpp:212:37 in 
MS: 0 ; base unit: 0000000000000000000000000000000000000000
0xa,0x5c,0x0,0x1,0x0,0x0,0x10,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,
\x0a\\\x00\x01\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00
artifact_prefix='./'; Test unit written to ./crash-f5bef9785939348fdc554918c368ce0977d8fd7a
Base64: ClwAAQAAEAAAAAAAAAAA
```

```
INFO: Seed: 2191391191
INFO: Loaded 1 modules   (515777 inline 8-bit counters): 515777 [0x55f8604c4498, 0x55f860542359), 
INFO: Loaded 1 PC tables (515777 PCs): 515777 [0x55f860542360,0x55f860d20f70), 
INFO:      510 files found in /tmp/cirrus-ci-build/ci/scratch/qa-assets/fuzz_seed_corpus/netbase_dns_lookup
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: seed corpus: files: 510 min: 1b max: 4042b total: 273870b rss: 117Mb
netbase.cpp:212:37: runtime error: implicit conversion from type 'int' of value -2147483632 (32-bit, signed) to type 'uint16_t' (aka 'unsigned short') changed the value to 16 (16-bit, unsigned)
    #0 0x55f85dbd91cc  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x338e1cc)
    #1 0x55f85cfd0f5f  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x2785f5f)
    #2 0x55f85ccebeb7  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x24a0eb7)
    #3 0x55f85e498067  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x3c4d067)
    #4 0x55f85e497d15  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x3c4cd15)
    #5 0x55f85cbf1851  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a6851)
    #6 0x55f85cbf0f95  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a5f95)
    #7 0x55f85cbf38b7  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a88b7)
    #8 0x55f85cbf3c19  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23a8c19)
    #9 0x55f85cbe28ee  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23978ee)
    #10 0x55f85cc0b732  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x23c0732)
    #11 0x7f05b2c830b2  (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #12 0x55f85cbb768d  (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/fuzz/fuzz+0x236c68d)
SUMMARY: UndefinedBehaviorSanitizer: implicit-signed-integer-truncation netbase.cpp:212:37 in 
MS: 0 ; base unit: 0000000000000000000000000000000000000000
0xa,0x5c,0x0,0x1,0x0,0x0,0x10,0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,
\x0a\\\x00\x01\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00
artifact_prefix='./'; Test unit written to ./crash-f5bef9785939348fdc554918c368ce0977d8fd7a
Base64: ClwAAQAAEAAAAAAAAAAA
```

**Comment by hebasto** [comment]
I hope it is fixed in #21328.

**Comment by jonatack** [comment]
Yes :+1: hopefully https://github.com/bitcoin/bitcoin/pull/21328#issuecomment-800523781 should get it. I hit this issue in the fuzz CI there.
